### PR TITLE
Fix status page bug

### PR DESCRIPTION
**Pull request by Calvinp**
There was a bug causing a console error on the Status Page. Now there isn't.
There also aren't lint errors or `atomicDisplayItems` components on the Status Page anymore.

This was branched off of #1131 so merge that first (and to see a good diff).

cc @tpetr @kwm4385 @wolfd

### CODE REVIEW

**Comment by Calvinp** [comment]
This is ready to be reviewed then merged.

**Comment by kwm4385** [comment]
LGTM

File: SingularityUI/app/components/status/StatusList.jsx [hunk 1]
```diff
@@ -88,11 +79,10 @@ export default class StatusList extends React.Component {
 StatusList.propTypes = {
   header: React.PropTypes.string,
   data: React.PropTypes.arrayOf(React.PropTypes.shape({
-      component: React.PropTypes.func.isRequired,
-      value: React.PropTypes.number,
-      prop: React.PropTypes.object,
-      id: React.PropTypes.string,
-      className: React.PropTypes.string,
-      beforeFill: React.PropTypes.string
+    component: React.PropTypes.func.isRequired, // className -> node
```
Comment (tpetr) [inline SingularityUI/app/components/status/StatusList.jsx]: what does this comment mean?

Comment (Calvinp) [inline SingularityUI/app/components/status/StatusList.jsx]: Signature of the function.

**Comment by tpetr** [comment]
LGTM

File: SingularityUI/app/components/status/Breakdown.jsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export default class Breakdown extends React.Component {
               style={{height: `${d.percent}%`}}
               className={`chart__data-point chart-fill-${d.type}`}
               data-original-title={`${d.count} ${d.label}`}
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  />
```
Comment (wolfd) [inline SingularityUI/app/components/status/Breakdown.jsx]: Love this. Something in eslint does this.

Comment (Calvinp) [inline SingularityUI/app/components/status/Breakdown.jsx]: Yeah, the close brace was at column 499

File: SingularityUI/app/components/status/HostStates.jsx [hunk 3]
```diff
@@ -1,88 +1,54 @@
...
-            noPages
-            columnHeads={this.getTableHeaders()}
-            tableRows={this.getTableRows(this.props.hosts) || []}
-            />
-      </div>
+function HostStates(props) {
+  return (
+    <div>
+      <h2>Singularity scheduler instances</h2>
+      <SimpleTable
```
Comment (wolfd) [inline SingularityUI/app/components/status/HostStates.jsx]: We should probably just use this kind of table in cases of paginated stuff.

File: SingularityUI/app/components/status/StatusPage.jsx [hunk 4]
```diff
@@ -121,58 +120,46 @@ class StatusPage extends React.Component {
...
-          component: Link,
-          beforeFill: r.type,
-          prop: {
-            text: `${r.count} ${r.label} ${this.renderPercentage(r.count, model.totalRequests)}`,
-            url: r.link,
-            value: r.count,
-            id: r.type
-          }
+          component: (className) => (
+            <a href={request.link} className={className}>
```
Comment (wolfd) [inline SingularityUI/app/components/status/StatusPage.jsx]: Link?

File: SingularityUI/app/components/status/StatusPage.jsx [hunk 5]
```diff
@@ -121,58 +120,46 @@ class StatusPage extends React.Component {
...
-        component: Link,
-        beforeFill: t.type,
-        prop: {
-          text: `${t.count} ${t.label} ${this.renderPercentage(t.count, model.totalTasks)}`,
-          url: t.link,
-          value: t.count,
-          id: t.type
-        }
+        component: (className) => (
+          <a href={task.link} className={className}>
```
Comment (wolfd) [inline SingularityUI/app/components/status/StatusPage.jsx]: Link?

File: SingularityUI/app/components/status/StatusPage.jsx [hunk 6]
```diff
@@ -222,31 +209,31 @@ class StatusPage extends React.Component {
...
                 {
-                  component: Link,
-                  prop: {
-                    text: `${m.activeRacks} Active Racks`,
-                    url: 'racks/active',
-                    id: 'activeracks',
-                    value: m.activeRacks
-                  }
+                  component: (className) => (
+                    <a href="racks/active" className={className}>
```
Comment (wolfd) [inline SingularityUI/app/components/status/StatusPage.jsx]: Link?

File: SingularityUI/app/components/status/StatusPage.jsx [hunk 7]
```diff
@@ -222,31 +209,31 @@ class StatusPage extends React.Component {
...
                 {
-                  component: Link,
-                  prop: {
-                    text: `${m.decomissioningRacks} Decommissioning Racks`,
-                    url: 'racks/decommission',
-                    id: 'decomracks',
-                    value: m.decomissioningRacks
-                  }
+                  component: (className) => (
+                    <a href="racks/decommission" className={className}>
```
Comment (wolfd) [inline SingularityUI/app/components/status/StatusPage.jsx]: Link?

File: SingularityUI/app/components/status/StatusPage.jsx (hunk 5 above)
Comment (Calvinp) [inline SingularityUI/app/components/status/StatusPage.jsx]: I'm trying to get rid of the `atomicDisplayItems` components - that was on purpose.

File: SingularityUI/app/components/status/StatusPage.jsx [hunk 8]
```diff
@@ -222,31 +209,31 @@ class StatusPage extends React.Component {
...
                 {
-                  component: Link,
-                  prop: {
-                    text: `${m.deadRacks} Inactive Racks`,
-                    url: 'racks/inactive',
-                    id: 'inactiveracks',
-                    value: m.deadRacks
-                  }
+                  component: (className) => (
+                    <a href="racks/inactive" className={className}>
```
Comment (wolfd) [inline SingularityUI/app/components/status/StatusPage.jsx]: Link?

File: SingularityUI/app/components/status/Breakdown.jsx (hunk 2 above)
Comment (wolfd) [inline SingularityUI/app/components/status/Breakdown.jsx]: I've seen 8k to 9k ones. It's curious.

**Comment by wolfd** [comment]
LGTM when the `SimpleTable` is swapped for `UITable` and `react-router` `Link`s are added.

**Pull request merged**